Añade ESLint, plugin de ESLint para Cypress, corrige errores y separa en modulos `index.js`

MalconH opened the pull request.

MalconH committed:
Agrega ESLint

Se agrega a las dev dependencies ESLint con la styleguide de AirBnb
Se agregan algunas reglas especiales:
- Se desactiva el linebreak-style porque da errores en Windows ya que
  espera un LF, y Windows utiliza CRLF
- Se habilita el uso del operador `++` dentro de los bucles for por
  costumbre de usarlo así en vez de += 1

MalconH committed:
Arregla problemas de estilo

Al introducir ESLint, se cambia la indentacion de 4 a 2 espacios y
se usan comillas simples en vez de dobles.

MalconH committed:
Refactoriza la manera de cargar pokemones

Anteriormente, a través de un for loop se obtenía cada pokemon, y luego
se creaban las cartas. Para que estas se crearan en orden, se usaba
un `await` ante cada llamada a `obtenerPokemon()`.
Dado que ESLint no permite awaits en loops (ver `no-await-in-loop`), lo
que se hace es:

1. Se precrean vacias las cartas porque queda mejor en la interfaz.
2. En vez de usar `await`, se utiliza el indice actual del loop que se encarga
de hacer las 20 llamadas a la API para los datos de cada pokemon se
cargue en el orden correcto en la carta que corresponda.

MalconH committed:
Separa el index en diferentes modulos

MalconH committed:
Arregla error al asertar sobra la img de cartas

Al introducir la nueva forma de cargar pokemones en el
commit b4b9a32241576f0255413e2b3a521b7903dc6b08, se cambio el texto
del atributo alt de cada img. Al asertar daba error ya que en los
tests seguía el texto anterior.

MalconH committed:
Modulariza `index.js`

Se separa la funcioanalidad de la página para mejor organización
y legibilidad del código.

Se modifica la regla de eslintrc ya que los scripts no cargaban
bien si no se les agregaba su extensión.

MalconH committed:
Añade plugin de Cypress para ESLint

MalconH committed:
Reordena orden funciones

Había funciones que estaban siendo llamadas antes de ser declaradas.
Si bien funcionaba, ESLint mostraba un error `no-use-before-define`

MalconH committed:
Borra variables sin utilizar

MalconH committed:
Arregla error de reasignación a un parametro

MalconH committed:
Arregla error de ESLint

Mostraba un error en bootstrap ya que no encontraba dicho objeto
en el código, pero aún así funcionaba el código, ya que viene cargado
el script de bootstrap por CDN.

MalconH committed:
Arregla error en nombre de propiedad

Estaba accediendo a la propiedad _element de los objetos Modal paro
tener el <elemento> de mi modal guardado en una variable, pero ESLint
no permite variables con `_` (ver def. no-underscore-dangle).

MalconH committed:
Cambia orden de declaracion de funciones

ESLint daba errores ya que habia funciones que se llamaban antes
de estar declaradas.

MalconH committed:
Arregla error de ESLint

MalconH committed:
Cambia de lugar explicacion del paginador

OLucho commented on src/utilidades.js:
Revisar nomenclatura de funciones,
Deben empezar con un verbo

OLucho commented on src/ui/modal.js:
```diff
@@ -0,0 +1,98 @@
+  const $modal = document.querySelector('.modal');
+
+  obtenerPokemon(idPokemon).then((pokemon) => {
+    // Desempaco el JSON a constantes:
```
estos comments son redundantes

OLucho commented on src/utilidades.js:
```diff
@@ -0,0 +1,12 @@
+}
+
+export function hectogramosAKG(hectogramo) {
+  return hectogramo / 10;
```
Por qué dividido 10? que significa?
Pasarlo a variable ese numero, evitar magic numbers

OLucho commented on src/utilidades.js:
```diff
@@ -0,0 +1,12 @@
+}
+
+export function decimetrosAM(decimetros) {
+  return decimetros / 10;
```
idem

MalconH committed:
Agrega ESLint como formateador por defecto

MalconH committed:
Refactoriza funciones por malas prácticas

En base a comentarios de @OLucho se mop
- Se modifican las funciones en `utilidades.js` que no empezaban con un
verbo en infinitivo como deberían.
- Se eliminan los magic numbers en algunas de ellas.
- Se elimina comentario redundantes en `/ui/modal.js`
